components/textfield/index.css

```diff
@@ -10,117 +10,510 @@ OF ANY KIND, either express or implied. See the License for the specific languag
+                              * -1);
+    pointer-events: none;
+    content: '';
+    border-style: solid;
     border-color: transparent;
-    transition: box-shadow var(--spectrum-global-animation-duration-100) ease-in-out, border-color var(--spectrum-global-animation-duration-100) ease-in-out;
+    border-width: 0;
+    border-radius: calc(var(--mod-textfield-corner-radius, var(--spectrum-textfield-corner-radius))
+                        + var(--mod-textfield-border-width, var(--spectrum-textfield-border-width))
+                        + var(--mod-textfield-focus-indicator-gap, var(--spectrum-textfield-focus-indicator-gap)));
+
+    /* place in same cell: input, focus indicator, and grows sizer */
+    grid-row: 2 / span 1;
+    grid-column: 1 / span 2;
   }
 
-  &.spectrum-Textfield--quiet {
-    &:after {
+  &.is-keyboardFocused {
+    /* focus indicator is focused state */
+    &::after {
+      border-color: var(--highcontrast-textfield-focus-indicator-color, var(--mod-textfield-focus-indicator-color, var(--spectrum-textfield-focus-indicator-color)));
+      border-width: var(--mod-textfield-focus-indicator-width, var(--spectrum-textfield-focus-indicator-width));
+    }
+  }
+
+  .is-readOnly & {
+    &::after {
+      border: 0;
+    }
+  }
+
+  &--quiet {
+    &::after {
+      inline-size: 100%;
+      block-size: calc(100% + var(--mod-textfield-focus-indicator-gap, var(--spectrum-textfield-focus-indicator-gap)));
+      margin-inline-start: 0;
+      margin-block-start: 0;
       border-radius: 0;
+      border-color: var(--highcontrast-textfield-focus-indicator-color, var(--mod-textfield-focus-indicator-color, var(--spectrum-textfield-focus-indicator-color)));
+    }
+  }
+
+  &--quiet.is-keyboardFocused {
+    &::after {
+      border-width: 0 0 var(--mod-textfield-focus-indicator-width, var(--spectrum-textfield-focus-indicator-width)) 0;
+    }
+  }
+
+  /* TODO - reevaluate icon styles when search field is migrated */
+  /* align search icon */
+  .spectrum-Textfield-icon {
+    display: block;
+    position: absolute;
+    top: 0;
+    bottom: 0;
+    margin-block: auto;
+
+    color: var(--highcontrast-textfield-text-color-default, var(--mod-textfield-text-color-default, var(--spectrum-textfield-text-color-default)));
+
+    &.spectrum-Search-icon {
+      right: auto;
+    }
+    
+    .spectrum-Textfield.is-disabled & {
+      color: var(--highcontrast-textfield-text-color-disabled, var(--mod-textfield-text-color-disabled, var(--spectrum-textfield-text-color-disabled)));
+    }
+
+    .spectrum-Textfield--quiet & {
+      color: var(--highcontrast-textfield-text-color-default, var(--mod-textfield-text-color-default, var(--spectrum-textfield-text-color-default)));
+    }
+
+    .spectrum-Textfield--quiet.is-disabled & {
+      color: var(--highcontrast-textfield-text-color-disabled, var(--mod-textfield-text-color-disabled, var(--spectrum-textfield-text-color-disabled)));
+    }
+
+    .spectrum-Textfield.is-readOnly & {
```

Comment:
So the `<div class="spectrum-Textfield">` gets this class but does not get the attributes of `readonly` or `disabled` because it's a div. It would be good to keep the class for things like the icon, helptext, etc. that are outside of the input but need to know whether it's disabled or not. We could use sibling selectors but that could start to get hairy. Thoughts?